src/systems/input.rs

```diff
@@ -9,17 +10,55 @@ use bevy::prelude::*;
 // Player 2 -> UP -> 'W' Key
 // Player 2 -> DOWN -> 'S' Key
 pub fn keyboard_input_system(
-    mut players: Query<(&mut Transform, &Player)>,
+    mut commands: Commands,
+    mut players_query: Query<(&mut Transform, Entity, Option<&mut InputTimer>, &Player)>,
+    time: Res<Time>,
     key: Res<Input<KeyCode>>,
     window: Res<WindowDescriptor>,
 ) {
-    for (mut transform, player) in players.iter_mut() {
+    for (mut transform, entity, input_timer, player) in players_query.iter_mut() {
         match player.player_type {
             PlayerType::Left => {
                 if key.pressed(KeyCode::W) {
-                    transform.translation.y += PADDLE_VELOCITY
+                    if let Some(mut t) = input_timer {
+                        t.timer.tick(time.delta());
+                        let elapsed = t.timer.elapsed_secs();
+                        let d_v = get_translation(elapsed);
+                        let v = d_v * PADDLE_VELOCITY;
+                        transform.translation.y += v
+                    } else {
+                        let mut timer = Timer::new(Duration::from_secs_f32(1.0), false);
+                        commands.entity(entity).insert(InputTimer {
+                            timer: timer.clone(),
+                        });
+                        timer.tick(time.delta());
+                        let elapsed = timer.elapsed_secs();
+                        let d_v = get_translation(elapsed);
+                        let v = d_v * PADDLE_VELOCITY;
+                        transform.translation.y += v
+                    }
                 } else if key.pressed(KeyCode::S) {
-                    transform.translation.y -= PADDLE_VELOCITY
+                    if let Some(mut t) = input_timer {
+                        t.timer.tick(time.delta());
+                        let elapsed = t.timer.elapsed_secs();
+                        let d_v = get_translation(elapsed);
+                        let v = d_v * PADDLE_VELOCITY;
+                        transform.translation.y -= v
```

Comment:
There must be a way to avoid rewriting this code if the only difference is a `+` versus a `-`
Maybe...
```
transform.translation.y += v * match (key.pressed(KeyCode::W), key.pressed(KeyCode::S)) {
    (True, False) => 1,
    (False, True) => -1,
    _ => 0
}